TELCODOCS-801: Add the Contribution Guide and related files and folder (#3)
hybrid-cloud-patterns/newdocs: modules/doc_guidelines.adoc

[id="contributing-to-docs-doc-guidelines"]
= Documentation guidelines

:description: Documentation guidelines for contributing to the Hybrid Cloud Patterns Docs
:imagesdir: ../images

== General guidelines
When authoring content, follow these style guides:

* link:https://redhat-documentation.github.io/supplementary-style-guide/[Red Hat Supplementary Style Guide]
* link:https://www.ibm.com/docs/en/ibm-style[IBM Style Guide], especially link:https://www.ibm.com/docs/en/ibm-style?topic=word-usage[word usage]
+

> NOTE: Use your Red Hat e-mail when asked for IBMid

Reviewer: Minor suggestion. Also this note wouldn't apply to non-Red Hatters and non-IBMers, right? Maybe we should clarify that.

```suggestion
NOTE: Use your Red Hat e-mail when asked for your IBM ID.
```

Author: Reverting to `IBMid` since that's how IBM refers to it

.Examples
* `designing-guided-decision-tables.adoc` (Assembly of guided decision table modules)
* `guided-decision-tables.adoc` (Concept module)
* `creating-guided-decision-tables.adoc`` (Procedure module for creating)
* `guided-decision-table-examples.adoc` (Reference module with examples)

=== Content type attributes

Each `.adoc` file must contain a `:_content-type:` attribute in its metadata that indicates its file type. This information is used by some publication processes to sort and label files.

Add the attribute from the following list that corresponds to your file type:

* `:_content-type: ASSEMBLY`
* `:_content-type: CONCEPT`
* `:_content-type: PROCEDURE`
* `:_content-type: REFERENCE`

See, xref:doc_guidelines.adoc#assembly-file-metadata[Assembly file metadata] and xref:doc_guidelines.adoc#module-file-metadata[Module file metadata].

== Naming conventions for directories

Use lowercase. For directory with a multiple-word name, use lowercase separated by dash, for example `multicloud-gitops`.

== Language and grammar
Consider the following guidelines:
* Use present tense.
* Use active voice.

> * Use second person (you).

Reviewer: Minor suggestion:

```suggestion
* Use second person perspective (you, your).
```

> * Avoid first person (I, we, us).

Reviewer: Minor suggestion:
```suggestion
* Avoid first person perspective (I, we, us).
```

* Be gender neutral.
* Use the appropriate tone.
* Write for a global audience.

== Titles and headings
Use sentence-style capitalization in all titles and section headings. Ensure that titles focus on customer tasks instead of the product.

> For assemblies and task modules, use a gerund form in headings, such as:

Reviewer: Might be an idea to use the consistent module terminology.

```suggestion
For assemblies and procedure modules, use a gerund form in headings, such as:
```

//To use a discrete heading, just add `[discrete]` to the line before your unique ID. For example:

//----
//[discrete]
//[id="managing-authorization-policies_{context}"]
//== Managing authorization policies
//----

== Writing assemblies

For more information about forming assemblies, see the
link:https://redhat-documentation.github.io/modular-docs/#forming-assemblies[Red Hat modular docs reference guide] and the link:https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc[assembly template].

[id="assembly-file-metadata"]
=== Assembly file metadata
Every assembly file should contain the following metadata at the top, with no line spacing in between, except where noted:

----
:_content-type: ASSEMBLY                                        <1>
[id="<unique-heading-for-assembly>"]                            <2>
= Assembly title                                                <3>
include::_common-docs/common-attributes.adoc[]                   <4>
:context: <unique-context-for-assembly>                         <5>
                                                                <6>

> toc::[]                                                         <7>

Reviewer: Using the correct TOC attribute syntax:

```suggestion
:toc:                                                           <7>
```

:_content-type: REFERENCE
[id="cli-basic-commands_{context}"]
= Basic CLI commands
----

[id="attribute-files"]
== Attribute files

AsciiDoc attributes are variables you can use in common files to:

* avoid hard-coding brand-specific information,

* share content between multiple brands more easily.

All attribute files must be placed in the a separate attributes file. For example, `common-docs` directory.

It is acceptable to group related attributes in the `common-attributes.adoc` file under a comment, as shown in the following example:

----
//ACM
rh-rhacm-product: Red Hat Advanced Cluster Management (RHACM)
:rh-shortname: RHACM
//GitOps
:gitops-product: Red Hat OpenShift GitOps
:gitops-shortname: GitOps
----

For more information on attributes, see link: https://docs.asciidoctor.org/asciidoc/latest/key-concepts/#attributes.

== Formatting
Use the following links to refer to AsciiDoc markup and syntax.

* https://redhat-documentation.github.io/asciidoc-markup-conventions/[AsciiDoc Mark-up Quick Reference for Red Hat Documentation]

* https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/[AsciiDoc Syntax Quick Reference]

> If you are graduating to AsciiDoc from Markdown, see the https://docs.asciidoctor.org/asciidoc/latest/asciidoc-vs-markdown/#comparison-by-example[AsciiDoc to Markdown syntax comparison by example].

Reviewer: No suggestion, just want to say great resource to add here!
